controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -111,13 +117,35 @@ func (r *NamespacePoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *NamespacePoolReconciler) getPoolStatus(ctx context.Context, pool crd.NamespacePool) (map[string]int, error) {
+func (r *NamespacePoolReconciler) handleErrorNamespaces(ctx context.Context, errNamespaceList []string) error {
+	for _, nsName := range errNamespaceList {
+		r.Log.Info("Deleting namespace", "ns-name", nsName)
+		err := DeleteNamespace(ctx, r.Client, nsName)
+		if err != nil {
+			r.Log.Error(err, fmt.Sprintf("Error deleting namespace: %s", nsName))
+			return fmt.Errorf("handleErrorNamespace error: Couldn't delete namespace: %v", err)
+		}
+
+		r.Log.Info("Removing prometheus-operator associated with", "ns-name", nsName)
+		err = DeletePrometheusOperator(ctx, r.Client, nsName)
+		if err != nil {
+			r.Log.Error(err, fmt.Sprintf("Error deleting prometheus.%s", nsName))
+			return fmt.Errorf("handleErrorNamespace error: Couldn't delete prometheus operator: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *NamespacePoolReconciler) getPoolStatus(ctx context.Context, pool crd.NamespacePool) (map[string]int, []string, error) {
 	nsList := core.NamespaceList{}
+	var errNamespaceList []string
```

Comment:
I think we should make an empty list of strings here because this could return a nil if it hasn't been initialized.